egs/aishell/s10/chain/tdnnf_layer.py

```diff
@@ -137,6 +159,35 @@ def forward(self, x):
         return x
 
 
+class TDNN(nn.Module):
+    '''
+    This class implements the following topology in kaldi:
+      relu-batchnorm-dropout-layer name=tdnn1 dropout-per-dim-continuous=true dim=1024
+    '''
+
+    def __init__(self, input_dim, hidden_dim):
+        super().__init__()
+        # affine conv1d requires [N, C, T]
+        self.affine = nn.Conv1d(in_channels=input_dim,
+                                out_channels=hidden_dim,
+                                kernel_size=1)
+
+        # tdnn1_batchnorm requires [N, C, T]
+        self.batchnorm = nn.BatchNorm1d(num_features=hidden_dim,
+                                              affine=False)
+        
+        self.dropout = SharedDimScaleDropout(dim=2)
+
+    def forward(self, x, dropout=0.):
+        # input x is of shape: [batch_size, feat_dim, seq_len] = [N, C, T]
+        x = self.affine(x)
+        x = F.relu(x)
+        x = self.batchnorm(x)
+        x = self.dropout(x, alpha=dropout)
```

Comment:
Add a guard here:

```python
if self.training:
  x = self.dropout(x, alpha=dropout)
```

The same place for the following `FactorizedTDNN`.

Reply:
self.training is in dropout module forward function,
```
if self.training and alpha > 0.:
```
Should I change it?

Reply:
I see. Just keep it unchanged.